Create db and kube web handlers

kimlisa opened the pull request:
part of https://github.com/gravitational/teleport/issues/6466
part of https://github.com/gravitational/teleport/issues/5086

#### Description
- Create database and kubernetes handlers
- Define database and kubernetes access rules

kimlisa committed:
Make db and kube web handlers

r0mant commented on lib/web/apiserver.go:
```diff
@@ -305,6 +305,12 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*RewritingHandler, error) {
 	h.GET("/webapi/sites/:site/context", h.WithClusterAuth(h.getUserContext))
 
+	// Database access handlers.
+	h.GET("/webapi/sites/:site/databases", h.WithClusterAuth(h.siteDatabasesGet))
```
"site" is obsolete terminology - ok to keep it in handler paths but let's name functions `clusterDatabasesGet` and `clusterKubesGet`.

kimlisa replied:
renamed

r0mant commented on lib/web/ui/server.go:
```diff
@@ -96,3 +97,96 @@ func MakeServers(clusterName string, servers []services.Server) []Server {
+	Desc string `json:"desc"`
+	// Procotol is the database description.
+	Procotol string `json:"protocol"`
+	// Type is the database type, self-hosted or AWS RDS.
```
Nit: Instead of "or AWS RDS" I would just say "or cloud-hosted" which would cover RDS, Redshift, Cloud SQL and anything we'll add in future.

r0mant commented on lib/web/ui/usercontext.go:
```diff
@@ -66,6 +67,10 @@ type userACL struct {
 	// AppServers defines access to application servers
 	AppServers access `json:"appServers"`
+	// DbServers defines access to database servers.
+	DbServers access `json:"dbServers"`
```
Nit: `DBServers` for consistency with how it's capitalized in other places.

kimlisa committed:
Address CR
- rename handler name prefix "site" to "cluster"
- rename DbServers to DBServers
- removed URI field b/c it is not needed in front
- update test

kimlisa committed:
Fix spelling

awly commented on lib/web/ui/server.go:
```diff
@@ -96,3 +97,93 @@ func MakeServers(clusterName string, servers []services.Server) []Server {
+
+// MakeKubes creates ui kube objects and returns a list..
+func MakeKubes(clusterName string, servers []types.Server) []Kube {
+	uiKubeClusters := []Kube{}
```
```suggestion
	uiKubeClusters := make([]Kube, 0, len(servers))
```

kimlisa replied:
i didn't make this change, b/c the len is determine by how many kube clusters there are in each server

awly commented on lib/web/ui/server.go:
```diff
@@ -96,3 +97,93 @@ func MakeServers(clusterName string, servers []services.Server) []Server {
+
+// MakeDatabases creates server database objects.
+func MakeDatabases(clusterName string, servers []types.DatabaseServer) []Database {
+	uiServers := []Database{}
```
```suggestion
	uiServers := make([]Database, 0, len(servers))
```

kimlisa replied:
revised

russjones reviewed (approved):
Bot.

kimlisa committed:
Assign len to dbServers and rename tests
